### PR TITLE
Deprecate class 'MasterService' and create alternative class 'ClusterManagerService'

### PR DESCRIPTION
**Pull request by tlfeng**
### Description
To support inclusive language, the `master` terminology is going to be replaced by `cluster manager` in the code base.

- Deprecate class `MasterService` and create alternative class `ClusterManagerService`.
- Add a unit test to validate the method `ClusterService.getMasterService()` can still return an object in type `MasterService`.
- Rename all the existing references of `MasterService` to `ClusterManagerService`, and rename the local variable names.
- Deprecate public methods `ClusterServiceUtils.createMasterService(...)` and create alternative methods `createClusterManagerService(...)`

Note:
The class `ClusterManagerService` is a subclass of `MasterService`, the inheritance relationship is opposite from most of the other classes with `Master` in the name (that covered by issue https://github.com/opensearch-project/OpenSearch/issues/1684).
The reason is: 
There is a public method that has return value in type `MasterService`,
https://github.com/opensearch-project/OpenSearch/blob/388c80ad94529b1d9aad0a735c4740dce2932a32/server/src/main/java/org/opensearch/cluster/service/ClusterService.java#L221
And in the code for Performance Analyzer plugin, there is a local variable in type `MasterService`:
https://github.com/opensearch-project/performance-analyzer/blob/5ee4809ac1cda6517ed871aeb12c6635203e7f1d/src/main/java/org/opensearch/performanceanalyzer/collectors/MasterServiceEventMetrics.java#L219
If making the old class `MasterService` a subclass of the new class `ClusterManagerService`, the above usage will be broken.
Reversing the inheritance relationship, I'm able to keep the backwards compatibility of the method `getMasterService()` while deprecating the class `MasterService` and encourage using a new class `ClusterManagerService`.
 
### Issues Resolved
A part of issue https://github.com/opensearch-project/OpenSearch/issues/3543 and https://github.com/opensearch-project/OpenSearch/issues/3544#issuecomment-1190855300
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/framework/src/test/java/org/opensearch/cluster/service/FakeThreadPoolMasterServiceTests.java [hunk 1]
```diff
@@ -84,21 +84,21 @@ public void testFakeClusterManagerService() {
         doAnswer(invocationOnMock -> runnableTasks.add((Runnable) invocationOnMock.getArguments()[0])).when(executorService).execute(any());
         when(mockThreadPool.generic()).thenReturn(executorService);
 
-        FakeThreadPoolMasterService masterService = new FakeThreadPoolMasterService(
+        FakeThreadPoolMasterService clusterManagerService = new FakeThreadPoolMasterService(
```
Comment (kotwanikunal) [inline test/framework/src/test/java/org/opensearch/cluster/service/FakeThreadPoolMasterServiceTests.java:87]: Does the class `FakeThreadPoolMasterService` need to be renamed as well?

**Review by kotwanikunal — COMMENTED** [review]
LGTM apart from the `FakeThreadPoolMasterService` comment.

File: test/framework/src/test/java/org/opensearch/cluster/service/FakeThreadPoolMasterServiceTests.java (hunk 1 above)
Comment (tlfeng) [reply test/framework/src/test/java/org/opensearch/cluster/service/FakeThreadPoolMasterServiceTests.java:87]: @kotwanikunal Thanks for your review. I forgot to write a note for this. 😂
I'm not going to rename this class in this PR. I plan to use the way of dealing with other classes on the 3 classes: `FakeThreadPoolMasterService` `BlockMasterServiceOnMaster` `BusyMasterServiceDisruption`.
One PR to rename directly, and one PR to create a new class with the old name then make it a subclass of the class in new name.

Comment (kotwanikunal) [reply test/framework/src/test/java/org/opensearch/cluster/service/FakeThreadPoolMasterServiceTests.java:87]: 👍 . Hopefully you have a task to keep track of these 🙂

**Review by kotwanikunal — APPROVED** [review]

**Review by andrross — APPROVED** [review]

File: test/framework/src/test/java/org/opensearch/cluster/service/FakeThreadPoolMasterServiceTests.java (hunk 1 above)
Comment (tlfeng) [reply test/framework/src/test/java/org/opensearch/cluster/service/FakeThreadPoolMasterServiceTests.java:87]: 😁 I listed the remaining items in this issue https://github.com/opensearch-project/OpenSearch/issues/3543#issuecomment-1189593243, and I will check it before wrapping up the work.

**Comment by tlfeng** [comment]
@andrross Thanks for your time on this PR! 👍 
I will hold on merging this PR until PR https://github.com/opensearch-project/OpenSearch/pull/4006 got merged and backported.

**Pull request merged**